src/Microsoft.DotNet.ImageBuilder/src/OctokitClientFactory.cs

```diff
@@ -74,14 +74,15 @@ information about the App itself.
                 var appClient = CreateClient(appCredentials);
                 var appInfo = await GetCurrentAppInfoAsync(appClient.Credentials);
 
-                Installation? appInstallation = appInfo.Installations.FirstOrDefault()
-                    ?? throw new InvalidOperationException(
-                        $"""
-                        No installations found for GitHub App.
-                        Ensure the App is installed on the organization you are trying to access.
-
-                        {appInfo}
-                        """);
+                Installation? appInstallation;
+
+                if (authOptions.InstallationId is long providedId) {
+                    appInstallation = appInfo.Installations.FirstOrDefault(i => i.Id == providedId)
+                        ?? throw new InvalidOperationException($"No installation found with ID {providedId}.");
+                } else {
+                    appInstallation = appInfo.Installations.SingleOrDefault()
+                        ?? throw new InvalidOperationException("Expected exactly one installation for GitHub App but found none or multiple. Provide an installation ID explicitly.");
```

Comment:
Nit: line over 120 chars
```suggestion
                        ?? throw new InvalidOperationException("Expected exactly one installation for GitHub App but "
                            + "found none or multiple. Provide an installation ID explicitly.");
```